include/Common.h

```diff
@@ -16,6 +17,9 @@ namespace Common {
 
     int RandomInt(int min, int max);
 
+    template <typename T>
+    T RandomEvent(int denominator, std::vector<int> numerators, std::vector<T> outcomes);
```

Comment:
On second thought, I think a better function signature would be `T RandomEvent(std::vector<int> weights, std::vector<T> outcomes)`. The denominator parameter is kind of redundant since you can just take the sum of all the weights (numerators) to get the denominator.

For example:
```c++
int GiantCrab::ReturnRiskAttackDamage() {
    // 4/6 chance of doing 0 damage
    // 2/6 chance of doing 20 damage

    vector<int> weights{4, 2};
    vector<int> outcomes{0, 20};
    return RandomEvent<int>(weights, outcomes);
}
```

Reply:
I also updated the signature